File: OneCodeTeam/Increase-the-size-of-the-ASPNET-Temporary-Folder-25342/[VB]-Increase the size of the ASP.NET Temporary Folder in Windows Azure role instance/VB/VBAzureIncreaseTempFolderSize/IncreaseAspnetTempFolderSize/WebRole.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports Microsoft.WindowsAzure
Imports Microsoft.WindowsAzure.Diagnostics
Imports Microsoft.WindowsAzure.ServiceRuntime
Imports Microsoft.Web.Administration

Public Class WebRole
    Inherits RoleEntryPoint

    Public Overrides Function OnStart() As Boolean
        ' For information on handling configuration changes
        ' see the MSDN topic at http://go.microsoft.com/fwlink/?LinkId=166357.

        ' Get the location of the AspNetTemp1GB resource     
        Dim aspNetTempFolder As Microsoft.WindowsAzure.ServiceRuntime.LocalResource = Microsoft.WindowsAzure.ServiceRuntime.RoleEnvironment.GetLocalResource("AspNetTemp1GB")

        ' Instantiate the IIS ServerManager     
        Dim iisManager As New ServerManager()

        ' Get the website.  Note that "_Web" is the name of the site in the ServiceDefinition.csdef, 
        ' so make sure you change this code if you change the site name in the .csdef     
        Dim app As Application = iisManager.Sites(RoleEnvironment.CurrentRoleInstance.Id + "_Web").Applications(0)

        ' Get the web.config for the site     
        Dim webHostConfig As Configuration = app.GetWebConfiguration()

        ' Get a reference to the system.web/compilation element     
        Dim compilationConfiguration As ConfigurationSection = webHostConfig.GetSection("system.web/compilation")

        ' Set the tempDirectory property to the AspNetTemp1GB folder     
        compilationConfiguration.Attributes("tempDirectory").Value = aspNetTempFolder.RootPath

        ' Commit the changes     
        iisManager.CommitChanges()

        Return MyBase.OnStart()
    End Function

End Class

File: OneCodeTeam/How-to-increase-the-size-of-the-ASPNET-Temporary-23689/[C#]-How to increase the size of the ASP.NET Temporary Folder in Azure/C#/CSAzureIncreaseTempFolderSize/IncreaseAspnetTempFolderSize/WebRole.cs
/****************************** Module Header ******************************\
Module Name:  WebRole.cs
Project:      IncreaseAspnetTempFolderSize
Copyright (c) Microsoft Corporation.

Increase the size of Asp.net Temporary folder in Azure. 
 
This source is subject to the Microsoft Public License.
See http://www.microsoft.com/en-us/openness/licenses.aspx#MPL
All other rights reserved.

THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND, 
EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED 
WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
\***************************************************************************/
using System;
using Microsoft.WindowsAzure;
using Microsoft.WindowsAzure.Diagnostics;
using Microsoft.WindowsAzure.ServiceRuntime;
using Microsoft.Web.Administration;

namespace IncreaseAspnetTempFolderSize
{
    public class WebRole : RoleEntryPoint
    {
        public override bool OnStart()
        {
            // For information on handling configuration changes
            // see the MSDN topic at http://go.microsoft.com/fwlink/?LinkId=166357.
            
            // Get the location of the AspNetTemp1GB resource     
            Microsoft.WindowsAzure.ServiceRuntime.LocalResource aspNetTempFolder = 
                Microsoft.WindowsAzure.ServiceRuntime.RoleEnvironment.GetLocalResource("AspNetTemp1GB");   
    
            //Instantiate the IIS ServerManager     
            ServerManager iisManager = new ServerManager();     

            // Get the website.  Note that "_Web" is the name of the site in the ServiceDefinition.csdef, 
            // so make sure you change this code if you change the site name in the .csdef     
            Application app = iisManager.Sites[RoleEnvironment.CurrentRoleInstance.Id + "_Web"].Applications[0];    
 
            // Get the web.config for the site     
            Configuration webHostConfig = app.GetWebConfiguration();     

            // Get a reference to the system.web/compilation element     
            ConfigurationSection compilationConfiguration = webHostConfig.GetSection("system.web/compilation");  
   
            // Set the tempDirectory property to the AspNetTemp1GB folder     
            compilationConfiguration.Attributes["tempDirectory"].Value = aspNetTempFolder.RootPath;   
  
            // Commit the changes     
            iisManager.CommitChanges();

            return base.OnStart();
        }
    }
}
